### PR TITLE
chore(react): Upgrade to React 18

### PR DESCRIPTION
**Pull request by damassi**
The type of this PR is: **Feat**

This PR solves [DIA-952]

### Description

This updates our app react dependencies to 18. 

Not so bad. Lots of laborious test and type updates, and was able to get all of our main type libraries to current, some which were a few major version behind. 

There are a few spots that I've `@ts-ignored` with REACT_18_UPGRADE. Not many tho. 

Along the way updated palette to 18 too, as a major version bump. 

This should just work! Will merge to staging tomorrow and then run through Integrity and fast follow if any issues are encountered. 

cc @artsy/diamond-devs 

[DIA-952]: https://artsyproduct.atlassian.net/browse/DIA-952?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/Artist/Routes/AuctionResults/Components/AuctionFilters/AuctionHouseFilter.tsx [hunk 1]
```diff
@@ -38,15 +40,18 @@ export const AuctionHouseFilter: React.FC<React.PropsWithChildren<unknown>> = ()
...
               onSelect: (selected: boolean) => {
                 toggleSelection(selected, name)
               },
               my: 1,
               selected: organizations?.includes(name),
               testID: `organization-filter-${name}`,
             }
-            return <Checkbox {...props}>{displayName}</Checkbox>
+            return (
+              <Checkbox key={index} {...props}>
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/Apps/Artist/Routes/AuctionResults/Components/AuctionFilters/AuctionHouseFilter.tsx:51]: keys typically have their own internal scope, so typically safe to use index (or rather, you'll know when you know you can't) 👍

[the remark this answers is not in the document]
Comment (dzucconi) [reply src/Apps/Artist/Routes/AuctionResults/Components/AuctionFilters/AuctionHouseFilter.tsx:51]: Which is to say this is fine

**Comment by damassi** [comment]
Re Enzyme: Yeah, so many tests still use that flow. I wish we had more resources to migrate. I'm personally waiting until LLMs get more accurate, or GitHub releases better repository tools. At that point we can skip all the work and simply prompt it.

**Pull request merged**